### ISSUE TITLE
HikariCP doesn't recover from connection marked as broken because of SQLSTATE(08006), ErrorCode(0)

### ISSUE THREAD

**Issue by victropolis** [brettwooldridge/HikariCP#627]
Yesterday we saw in production where HikariCP didn't seem to be able to recover from 

```
connection marked as broken because of SQLSTATE(08006), ErrorCode(0).  
```

We had four instances of our application running on four separate tomcat containers and bounced them at the same time.  All four had problems connecting to the database; some with the above error and HikariCP didn't seem to be able to recover from it.

I had to start each container individually and then it worked fine.  Not sure if postgres was throttling the connections

**Comment by brettwooldridge** [comment]
@victropolis What is the postgres configured ``max_connections``?  What is ``tcp_keepalives_idle`` set to (I recommend 60-120 seconds).  Finally, what version of HikariCP?

**Comment by victropolis** [comment]
@brettwooldridge we're on HikariCP 2.4.4.  Are those max_connections and tcp_keepalives_idle settings on the postgres service?  Thanks!

**Comment by brettwooldridge** [comment]
@victropolis The referred to settings are postgres settings.  And as always, we recommend verifying the behavior still occurs on the latest available release (in this case, v2.4.6).

I recommend enabling debug logging (it's not too noisy), and trying to see if you can either reproduce the behavior.  The logs could be very informative.
